### PR TITLE
Add cut and legato, move sample trimming related functions to their own page

### PR DESCRIPTION
**Pull request by Asmatzaile**
There were a lot of functions that dealt with trimming samples and I felt there was the need to classify them in functions that trim samples in an absolute manner (sustain), functions that trim them relative to their length (begin, end, grain...) and functions that trim them relative to the events of the pattern (cut, legato).
I did that classification and I moved them to their own page. Is there a way I have to link the page so that it appears in the documentation?

### CODE REVIEW

**Comment by trespaul** [comment]
Hey! So, we're currently reconsidering the structure of the documentation (preview at https://tidal-doc.pages.dev, source at github.com/trespaul/tidal-doc), including the reference section, which I personally am of the opinion should reflect the modules as they are in the Tidal source code. We're following the ["Grand Unified Theory of Documentation"](https://documentation.divio.com/), so we're keeping guides and reference sections separate. Please take a look at the new site and let us know what you think! Specifically, do you think we should make a guide for sample trimming functions?

File: docs/reference/samplers.md [hunk 1]
```diff
@@ -97,45 +58,9 @@ d1 $ s "ab*16" # gain (range 0.8 1.3 $ sine)
...
-```
-
-This example is equivalent to:
-
-```haskell
-d1 $ slow 2 $ s "bev" # begin 0.2 # end 0.3 # legato 1
- ```
-
-## Sample effects
+These section presents effects change  that both the speed and the pitch of the samples.
```
Comment (trespaul) [inline docs/reference/samplers.md:63]: *This section

**Comment by Asmatzaile** [comment]
Nice restructuring!
I do like that now the reference reflects how the code is structured in hackage and the guides port the information that was in the previous documentation.
I do think there should be a guide for sample trimming functions, for a couple of reasons: a) I think these functions are very related conceptually, and it's therefore good that they are grouped together in a page. b) has some readily available information about changing 'legato' and 'cut' that I spent some time researching and could be of use to other classical musicians.
In fact, I have been thinking about how the functions could be better grouped, and I was getting made to make another PR with that (currently, 'samples' and 'sampling' are a little bit confusing).
For now, I would definitely create that new guide about sample trimming functions. Should I go ahead and do it?

**Comment by trespaul** [comment]
Since you've already made the page, I'm happy to add it to my fork and tag you for credit, or if you like, you can open a PR on that repo. Regarding the organisation of the functions: there has been some discussion about that; we're talking about it on the Discord, if you'd like to join, or if you don't have a discord account we can discuss on github somewhere.

**Comment by Asmatzaile** [comment]
Okay, thanks! Checked the discord, and just made a [PR in your repo](trespaul/tidal-doc#1)!

**Comment by yaxu** [comment]
> should reflect the modules as they are in the Tidal source code

We could also say vice-versa, as the organisation in source isn't necessarily perfect.

In the meantime shall we merge this PR?

**Comment by trespaul** [comment]
> We could also say vice-versa

Yes, you're right, I meant to say that they should be in sync :sweat_smile:

**Comment by cleary** [comment]
> In the meantime shall we merge this PR?

They're key functions imo, so :+1: from me

**Comment by yaxu** [comment]
Done, thanks!

**Pull request merged**